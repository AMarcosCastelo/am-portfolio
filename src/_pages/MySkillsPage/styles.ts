import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: ${({ theme }) => theme.palette.background};
  width: 100vw;
  height: 100vh;
  position: relative;
  display: flex;
  justify-content: space-evenly;
  align-items: center;
`

export const Main = styled.div`
  border: 2px solid ${({ theme }) => theme.palette.text.main};
  color: ${({ theme }) => theme.palette.text.main};
  background-color: ${({ theme }) => theme.palette.background.default};
  padding: 2rem;
  width: 30vw;
  height: 60vh;
  z-index: 3;
  line-height: 1.5;
  cursor: pointer;
  font-family: 'Ubuntu Mono', monospace;
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  &:hover {
    color: ${({ theme }) => theme.palette.background.default};
    background-color: ${({ theme }) => theme.palette.text.main};
  }
`

export const Title = styled.h2`
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: calc(1rem + 1vw);

  ${Main}:hover & {
    & > * {
      fill: ${({ theme }) => theme.palette.background.default};
    }
  }

  & > *:first-child {
    margin-right: 1rem;
  }
`

export const Description = styled.div`
  color: ${({ theme }) => theme.palette.text.main};
  font-size: calc(0.5rem + 1vw);
  padding: 0.5rem 0;

  strong {
    margin-bottom: 1rem;
    text-transform: uppercase;
  }

  ul,
  p {
    margin-left: 2rem;
  }

  ${Main}:hover & {
    color: ${({ theme }) => theme.palette.background.default};
  }
`
