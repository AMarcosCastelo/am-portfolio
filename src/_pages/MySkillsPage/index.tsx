import * as S from './styles'

const MySkillsPage = () => (
  <S.Wrapper>
    <h1>MySkillsPage</h1>
  </S.Wrapper>
)

export default MySkillsPage
