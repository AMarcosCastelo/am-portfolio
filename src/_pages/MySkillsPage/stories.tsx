import { Story, Meta } from '@storybook/react'
import MySkillsPage from '.'

export default {
  title: 'MySkillsPage',
  component: MySkillsPage
} as Meta

export const Default: Story = () => <MySkillsPage />
