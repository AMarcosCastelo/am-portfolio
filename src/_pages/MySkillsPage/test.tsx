import { render } from '@testing-library/react'

import MySkillsPage from '.'

describe('<MySkillsPage />', () => {
  it('should render the heading', () => {
    const { container } = render(<MySkillsPage />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
