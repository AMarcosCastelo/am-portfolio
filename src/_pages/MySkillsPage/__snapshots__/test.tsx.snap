// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MySkillsPage /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    MySkillsPage
  </h1>
</main>
`;
